### ISSUE TITLE
Several issues

### ISSUE THREAD

**Issue by sup3rock** [TheMody/GeneDiffusion#1]
Hi there,
I was wondering if this codebase is the one actually used to produce the paper results, or if it's an **untested** rewritten version.

Because I wanted to run some tests, but even just in the dataloader I noticed some problems:

1. When you normalise the data, you are not really subtracting the mean: https://github.com/TheMody/GeneDiffusion/blob/2d8544b484025f44987fd4987880d6cee1c34a66/utils/dataloader.py#L353
2. Even assuming that problem 1 does not exist, you are normalising the whole dataset and then splitting it into training and test splits. This is against best practice. You should first split and then normalise the test split using the statistics from the train split.

Thanks

**Comment by TheMody** [comment]
Hi there,
Thanks for taking a closer look at the code.
1.This code was actually used to produce the results reported in the paper and later rewritten slightly to make it cleaner, through this process some bugs could have accumulated, but should be very minor. The x-xmean seems to be a definite error, which will be fixed. Thanks again for pointing this out. Luckily the mean of the input data was already at 0 due to the PCA pre-processing as mentioned in the paper.
2.To make the pre-processing cleaner this will be added, but i do not expect any change in results. This point is very marginal at these dataset sizes. Still i will test if this changes any results.
Best,
